opencti-platform/opencti-graphql/src/utils/filtering/filtering-resolution.ts

```diff
@@ -217,11 +217,11 @@ export const resolveFiltersMapForUser = async (context: AuthContext, user: AuthU
 };
 
 export const convertFiltersToQueryOptions = async (filters: FilterGroup | null, opts: any = {}) => {
-  const { after, before, defaultTypes = [], field = 'updated_at', orderMode = 'asc' } = opts;
+  const { after, before, defaultTypes = [], extraFilters = [], field = 'updated_at', orderMode = 'asc' } = opts;
   const types = [...defaultTypes];
   let finalFilters = filters;
-  if (after || before) {
-    const filtersContent = [];
+  if (after || before || extraFilters.length > 0) {
+    const filtersContent = [...extraFilters];
```

Comment:
I'm a bit confused about this, how would it work with after and before ? I know in this current case extraFilters is not used with after or before, but what if it's used in the future ? what filter would it create ? would it be a "And" or "Or" condition between them ?

Reply:
For now its a AND condition that feat the needs i needs. Don't want to complexify the options before having a real use case.

Reply:
so it's an AND by default in filters ? I just find the code less easy to read when having an if condition between after before and extraFilters, since they are not related (after / before are related). I'm wondering if we really need this extraFilters param, couldn't we create the right FilterGroup object before calling this method ?

Reply:
Before and after are not directly related, thats extra options that can accumulate a first set of filters. Its and AND because these filters are use with

```
mode: FilterMode.And,
filters: filtersContent,
```

ExtraFilters options is the simplest way to add some filters at the firt level to accumulate with other options.